src/core/agent/services/signifyNotificationService.ts

```diff
@@ -225,11 +225,18 @@ class SignifyNotificationService extends AgentService {
         const [admit, sigs, aend] = await this.props.signifyClient
           .ipex()
           .admit(ourIdentifier.signifyName, "", notif.a.d, dt);
-        await this.props.signifyClient
+        const op = await this.props.signifyClient
           .ipex()
           .submitAdmit(ourIdentifier.signifyName, admit, sigs, aend, [
             exchange.exn.i,
           ]);
+        const pendingOperation = await this.operationPendingStorage.save({
```

Comment:
Can we add a check before we `admit` to make sure that the status is 0 (not revoked) and 1 after? Just in case we accidentally got 2 grant messages for the issuance, but it was never changed to revoked.

Reply:
We can only check the status after the operation is done so I think we should check this in `onSignifyOperationStateChanged`, after we submit admit

Reply:
Actually the grant message should have the ACDC embedded with the status. You don't need to call `credentials().get()`. The problem with having it where you have it now is that if we receive 2 revoke messages in a row by accident, we will have 2 revoked notifications